extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -0,0 +1,105 @@
+from conan.cli.args import common_graph_args, validate_common_graph_args
+from conan.cli.command import conan_command
+from conans.client.graph.graph import Node
+
+lFac = LicenseFactory()
+
+
+def package_type_to_component_type(pt: str) -> ComponentType:
+    if pt is "application":
+        return ComponentType.APPLICATION
+    else:
+        return ComponentType.LIBRARY
+
+
+def licenses(ids):
+    """
+    see https://cyclonedx.org/docs/1.4/json/#components_items_licenses
+    """
+    if ids is None:
+        return None
+    if not isinstance(ids, tuple):
+        ids = [ids]
+    return [LicenseChoice(license=lFac.make_from_string(i)) for i in ids]
+
+
+def package_url(node: Node) -> PackageURL:
+    """
+    Creates a PURL following https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#conan
+    """
+    return PackageURL(
+        type="conan",
+        name=node.conanfile.name,
+        version=node.conanfile.version,
+        qualifiers={
+            "prev": node.prev,
+            "rref": node.ref.revision,
+            "user": node.conanfile.user,
+            "channel": node.conanfile.channel,
+            "repository_url": node.remote.url if node.remote else None
+        })
+
+
+def create_component(n: Node) -> Component:
+    result = Component(
+        type=package_type_to_component_type(n.conanfile.package_type),
+        name=n.conanfile.name,
+        version=n.conanfile.version,
+        licenses=licenses(n.conanfile.license),
+        bom_ref=package_url(n).to_string(),
+        purl=package_url(n),
+        description=n.conanfile.description
+    )
+    if n.conanfile.homepage:
+        result.external_references.add(ExternalReference(
+            type=ExternalReferenceType.WEBSITE,
+            url=XsUri(n.conanfile.homepage),
+        ))
+    return result
+
+
+@conan_command(group="Recipe")
+def create_sbom(conan_api: ConanAPI, parser, *args):
+    """
+    creates an SBOM in CycloneDX 1.4 JSON format from a Conan graph JSON
+    """
+
+    # BEGIN COPY FROM conan: cli/commands/graph.py
+    common_graph_args(parser)
+    args = parser.parse_args(*args)
+    validate_common_graph_args(args)
+    cwd = os.getcwd()
+    path = conan_api.local.get_conanfile_path(args.path, cwd, py=None) if args.path else None
+    remotes = conan_api.remotes.list(args.remote) if not args.no_remote else []
+    overrides = eval(args.lockfile_overrides) if args.lockfile_overrides else None
+    lockfile = conan_api.lockfile.get_lockfile(lockfile=args.lockfile,
+                                               conanfile_path=path,
+                                               cwd=cwd,
+                                               partial=args.lockfile_partial,
+                                               overrides=overrides)
+    profile_host, profile_build = conan_api.profiles.get_profiles_from_args(args)
+    deps_graph = conan_api.graph.load_graph_consumer(path, args.name, args.version,
```

Comment:
Maybe a bit more exotic (not sure whether we should support the following syntax):
conan recipe:create-sbom --requires openssl/1.1.1t
will lead to
```
  File "[...]\lib\site-packages\conan\api\subapi\graph.py", line 23, in _load_root_consumer_conanfile
    if path.endswith(".py"):
AttributeError: 'NoneType' object has no attribute 'endswith'
```
Note that the following command works fine:
conan graph info --requires openssl/1.1.1t

Reply:
None value checked and test added